bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/RuleSupportScriptExtension.java

```diff
@@ -0,0 +1,198 @@
+        staticTypes.put("SimpleConditionHandler", SimpleConditionHandler.class);
+
+        staticTypes.put("ActionHandlerFactory", ScriptedActionHandlerFactory.class);
+        staticTypes.put("ConditionHandlerFactory", ScriptedConditionHandlerFactory.class);
+        staticTypes.put("TriggerHandlerFactory", ScriptedTriggerHandlerFactory.class);
+
+        staticTypes.put("Configuration", Configuration.class);
+        staticTypes.put("Action", Action.class);
+        staticTypes.put("Condition", Condition.class);
+        staticTypes.put("Trigger", Trigger.class);
+        staticTypes.put("Rule", Rule.class);
+        staticTypes.put("ModuleType", ModuleType.class);
+        staticTypes.put("ActionType", ActionType.class);
+        staticTypes.put("TriggerType", TriggerType.class);
+        staticTypes.put("ConfigDescriptionParameter", ConfigDescriptionParameter.class);
+
+        types.addAll(staticTypes.keySet());
+
+        types.add("AutomationManager");
+        types.add("RuleRegistry");
+        types.add("rules");
+
+        presets.put("RuleSupport",
+                Arrays.asList("Configuration", "Action", "Condition", "Trigger", "Rule", "ModuleType", "ActionType"));
+        presets.put("RuleSimple", Arrays.asList("ScriptedRule", "SimpleRule"));
+        presets.put("RuleFactories", Arrays.asList("ActionHandlerFactory", "ConditionHandlerFactory",
+                "TriggerHandlerFactory", "TriggerType", "ConfigDescriptionParameter"));
+    }
+
+    public void setRuleRegistry(RuleRegistry ruleRegistry) {
+        this.ruleRegistry = ruleRegistry;
+    }
+
+    public void setRuleProvider(ScriptedRuleProvider ruleProvider) {
+        this.ruleProvider = ruleProvider;
+    }
+
+    protected void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+        staticTypes.put("ItemRegistry", this.itemRegistry);
+        staticTypes.put("ir", this.itemRegistry);
+    }
+
+    protected void unsetItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = null;
+    }
+
+    @Override
+    public Collection<String> getDefaultPresets() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public Collection<String> getPresets() {
+        return presets.keySet();
+    }
+
+    @Override
+    public Collection<String> getTypes() {
+        return types;
+    }
+
+    @Override
+    public Object get(String scriptIdentifier, String type) {
+        Object obj = staticTypes.get(type);
+        if (obj != null) {
+            return obj;
+        }
+
+        HashMap<String, Object> objects = objectCache.get(scriptIdentifier);
+
+        if (objects == null) {
+            objects = new HashMap<>();
+            objectCache.put(scriptIdentifier, objects);
+        }
+
+        obj = objects.get(type);
+        if (obj != null) {
+            return obj;
+        }
+
+        if (type.equals("AutomationManager") || type.equals("RuleRegistry")) {
```

Comment:
Could you define all those strings as constants? This would prevent typos and simplify refactorings.